frontend/appflowy_flutter/lib/workspace/application/workspace/overview/overview_listener.dart

```diff
@@ -0,0 +1,104 @@
+import 'dart:async';
+import 'dart:typed_data';
+
+import 'package:appflowy/core/notification/folder_notification.dart';
+import 'package:appflowy_backend/dispatch/dispatch.dart';
+import 'package:appflowy_backend/log.dart';
+import 'package:appflowy_backend/protobuf/flowy-error/errors.pb.dart';
+import 'package:appflowy_backend/protobuf/flowy-folder/notification.pb.dart';
+import 'package:appflowy_backend/protobuf/flowy-folder/view.pb.dart';
+import 'package:appflowy_backend/protobuf/flowy-notification/subject.pb.dart';
+import 'package:appflowy_backend/rust_stream.dart';
+import 'package:dartz/dartz.dart';
+
+// parent view of overview block updated
+typedef ParentViewUpdateNotifier = void Function(ViewPB);
+
+typedef ChildViewsUpdateNotifier = void Function(ChildViewUpdatePB);
+
+class WorkspaceOverviewListener {
```

Comment:
What is the purpose of this class? If you want to receive updates for the current view and its child views, using ViewListener is sufficient in this case.

Reply:
The `ViewListener` class currently only monitors updates from its immediate view and its direct child views(i.e. first level child views). If `ViewListener` is used in the overview block, then the overview widget component lacks the functionality to listen for updates beyond the first level of child views. However, the `WorkspaceOverviewListener` class addresses this gap by enabling monitoring of changes in both its own view and all levels of child views in the hierarchy.